subs/pantry/src/Pantry/Storage.hs

```diff
@@ -497,40 +604,80 @@ getTreeForKey (P.TreeKey key) = do
     Nothing -> pure Nothing
     Just bid -> getBy $ UniqueTree bid
 
-loadPackageById
-  :: (HasPantryConfig env, HasLogFunc env)
-  => TreeId
-  -> ReaderT SqlBackend (RIO env) Package
+loadPackageById ::
+       (HasPantryConfig env, HasLogFunc env, HasProcessContext env)
+    => TreeId
+    -> ReaderT SqlBackend (RIO env) Package
 loadPackageById tid = do
-  mts <- get tid
-  ts <-
-    case mts of
-      Nothing -> error $ "loadPackageById: invalid foreign key " ++ show tid
-      Just ts -> pure ts
-  tree <- loadTreeByEnt $ Entity tid ts
-  key <- getBlobKey $ treeKey ts
-
-  mname <- get $ treeName ts
-  name <-
-    case mname of
-      Nothing -> error $ "loadPackageByid: invalid foreign key " ++ show (treeName ts)
-      Just (PackageName (P.PackageNameP name)) -> pure name
-
-  mversion <- get $ treeVersion ts
-  version <-
-    case mversion of
-      Nothing -> error $ "loadPackageByid: invalid foreign key " ++ show (treeVersion ts)
-      Just (Version (P.VersionP version)) -> pure version
-
-  cabalKey <- getBlobKey $ treeCabal ts
-  let ident = P.PackageIdentifier name version
-  let cabalEntry = P.TreeEntry cabalKey (treeCabalType ts)
-  pure Package
-    { packageTreeKey = P.TreeKey key
-    , packageTree = tree
-    , packageCabalEntry = cabalEntry
-    , packageIdent = ident
-    }
+    (mts :: Maybe Tree) <- get tid
+    ts <-
+        case mts of
```

Comment:
is there any point in splitting arrow and a case in 2 lines?

Reply:
@qrilka That's hindent there. Is it making it unreadable ?